nvidia_deeplearningexamples_gpunet.md

```diff
@@ -0,0 +1,133 @@
+---
+
+
+### Model Description
+GPUNets are a new family of deployment and production ready Convolutional Neural Networks from NVIDIA auto-designed to max out the performance of NVIDIA GPU and TensorRT. 
+
+Crafted by NVIDIA AI using novel Neural Architecture Search(NAS) methods, GPUNet demonstrates state-of-the-art inference performance up to 2x faster than EfficientNet-X and FBNet-V3. This notebook allows you to load and test all the the GPUNet model implementation listed in our [CVPR-2022 paper](https://arxiv.org/pdf/2205.00841.pdf). You can use this notebook to quickly load each one of listed models to perform inference runs.
+
+### Example
+In the example below the pretrained ***GPUNet-0*** model is loaded by default to perform inference on image and present the result. You can switch the default pre-trained model loading from GPUNet-0 to one of these: GPUNet-1, GPUNet-2, GPUNet-P0, GPUNet-P1, GPUNet-D1 or GPUNet-D2.
+### Install pre-requisites
+To run the example you need some extra python packages installed. These are needed for preprocessing images and visualization.
+```python
+!pip install validators matplotlib
+!pip install timm==0.5.4
+```
+
+```python
+import torch
+from PIL import Image
+import torchvision.transforms as transforms
+import numpy as np
+import json
+import requests
+import matplotlib.pyplot as plt
+import warnings
+
+warnings.filterwarnings('ignore')
+%matplotlib inline
+
+device = torch.device("cuda") 
+print(f'Using {device} for inference')
+
+if torch.cuda.is_available():
+    !nvidia-smi
+else:
+    torch.device("cpu")
+```
+
+### Load Pretrained model
+Loads NVIDIA GPUNet-0 model by default pre-trained on IMAGENET dataset. You can switch the default pre-trained model loading from GPUNet-0 to one of the following models listed below. 
+
+The model architecture is visible as output of the loaded model. For details architecture and latency info please refer to Figure#[6](https://drive.google.com/file/d/1pLA0OopYVRpYreIhQQKPdNauaaBEGITe/view) and Table#[3](https://drive.google.com/file/d/1pLA0OopYVRpYreIhQQKPdNauaaBEGITe/view) in the CVPR-2022 paper. 
+
+Please pick and choose one of the following pre-trained models:
+
+| TorchHub model | Description |
+| :----- | :----- |
+| `GPUNet-0` | GPUNet-0 has the fastest measured latency on GV100 |
+| `GPUNet-1` | GPUNet-1 has improved accuracy with one additional layer on GPUNet-0|
+| `GPUNet-2` | GPUNet-2 has higher accuracy with two additional layers on GPUNet-0 |
+| `GPUNet-P0` | GPUNet-P0 is the distilled model with higher accuracy than GPUNet-0 but similar latency|
+| `GPUNet-P1` | GPUNet-P1 is distilled model with even higher accuracy than GPUNet-1 but similar latency |
+| `GPUNet-D1` | GPUNet-D1 has the second highest accuracy amongst all GPUNets|
+| `GPUNet-D2` | GPUNet-D2 has the highest accuracy amongst all GPUNets |
+
+***Note***: Loaded model defaults to fp32 weights. To switch to fp16 gpunet model weights, set parameter ***model_math='fp16'***.
```

Comment:
not sure if that breaks the CI but `model_math='fp16'` is perhaps a bit neater?

Reply:
im not sure if i understand. you mean make fp16 the default?

Reply:
I mean using ` decorators instead of *

Reply:
Ha. Communication! :) For the last hour we were trying to wrap model object with fp16 halving **decorator** :P That's not the best idea btw, but anyway next commit will make the code much more clear :)

Reply:
My bad should have made my thinking clearer, i tend to assume everyone has an open access to my thoughts :)

Reply:
Fixed